chain/chain/src/chain.rs

```diff
@@ -200,6 +216,37 @@ impl OrphanBlockPool {
 
         ret
     }
+
+    pub fn get_orphans_within_depth(
+        &self,
+        prev_hash: CryptoHash,
+        target_depth: u64,
+    ) -> Vec<CryptoHash> {
+        let mut visited = HashSet::new();
+        let mut queue = vec![(prev_hash, 0)];
+        while let Some((prev_hash, depth)) = queue.pop() {
+            if depth == target_depth {
+                continue;
```

Comment:
noob question: isn't it better to break here?

Reply:
That's legit, I changed it to break here